### ISSUE TITLE
Change in re.escape output is not documented in 3.7 whatsnew

### ISSUE THREAD

**Issue by 874c7ef8-1afe-4082-ade7-d41534a84280** [python/cpython#74333]
BPO | [30147](https://bugs.python.org/issue30147)
--- | :---
Nosy | @ned-deily, @ezio-melotti, @bitdancer, @serhiy-storchaka, @wumpus
PRs | <li>python/cpython#5978</li><li>python/cpython#5979</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2018-03-04.23:24:40.235>
created_at = <Date 2017-04-23.18:39:56.804>
labels = ['3.8', 'type-bug', '3.7', 'docs']
title = 'Change in re.escape output is not documented in 3.7 whatsnew'
updated_at = <Date 2018-03-04.23:24:40.233>
user = 'https://github.com/wumpus'
```

bugs.python.org fields:
```python
activity = <Date 2018-03-04.23:24:40.233>
actor = 'ned.deily'
assignee = 'docs@python'
closed = True
closed_date = <Date 2018-03-04.23:24:40.235>
closer = 'ned.deily'
components = ['Documentation']
creation = <Date 2017-04-23.18:39:56.804>
creator = 'wumpus'
dependencies = []
files = []
hgrepos = []
issue_num = 30147
keywords = ['patch']
message_count = 5.0
messages = ['292177', '292179', '313226', '313227', '313228']
nosy_count = 7.0
nosy_names = ['ned.deily', 'ezio.melotti', 'mrabarnett', 'r.david.murray', 'docs@python', 'serhiy.storchaka', 'wumpus']
pr_nums = ['5978', '5979']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue30147'
versions = ['Python 3.7', 'Python 3.8']
```

</p></details>

**Comment by 874c7ef8-1afe-4082-ade7-d41534a84280** [comment]
The following script runs fine in python 3.6 and recently started failing the assertion in 3.7-dev and nightly

```py
import re

r = re.compile(re.escape('/foo'))
print(r)
print(r.pattern)
assert r.pattern.startswith('\\/')
```

**Comment by bitdancer** [comment]
re.escape was escaping too much previously, and this has been fixed in 3.7 (and only 3.7, because it may cause testing issues such as you have observed).  See bpo-29995.

A note needs to be added to the 'porting' section of whats new covering this.
